metadata-ingestion/src/datahub/sdk/chart.py

```diff
@@ -0,0 +1,322 @@
+        last_refreshed: Optional[datetime] = None,
+        chart_type: Optional[str] = None,
+        access: Optional[str] = None,
+        # Standard aspects.
+        subtype: Optional[str] = None,
+        owners: Optional[OwnersInputType] = None,
+        links: Optional[LinksInputType] = None,
+        tags: Optional[TagsInputType] = None,
+        terms: Optional[TermsInputType] = None,
+        domain: Optional[DomainInputType] = None,
+        input_datasets: Optional[List[Union[DatasetUrnOrStr, Dataset]]] = None,
+        extra_aspects: ExtraAspectsType = None,
+    ):
+        """Initialize a new Chart instance."""
+        urn = ChartUrn.create_from_ids(
+            platform=platform,
+            name=name,
+            platform_instance=platform_instance,
+        )
+        super().__init__(urn)
+        self._set_extra_aspects(extra_aspects)
+
+        self._set_platform_instance(platform, platform_instance)
+
+        # Initialize ChartInfoClass with all required arguments
+        chart_info = models.ChartInfoClass(
+            title=display_name or name,
+            description=description or "",
+            lastModified=models.ChangeAuditStampsClass(),  # TODO: this is required.
+        )
+
+        self._setdefault_aspect(chart_info)
+
+        # Set additional properties
+        if external_url is not None:
+            self.set_external_url(external_url)
+        if chart_url is not None:
+            self.set_chart_url(chart_url)
+        if custom_properties is not None:
+            self.set_custom_properties(custom_properties)
+        if last_refreshed is not None:
+            self.set_last_refreshed(last_refreshed)
+        if chart_type is not None:
+            self.set_chart_type(chart_type)
+        if access is not None:
+            self.set_access(access)
+        if subtype is not None:
+            self.set_subtype(subtype)
+        if owners is not None:
+            self.set_owners(owners)
+        if links is not None:
+            self.set_links(links)
+        if tags is not None:
+            self.set_tags(tags)
+        if terms is not None:
+            self.set_terms(terms)
+        if domain is not None:
+            self.set_domain(domain)
+        if last_modified is not None:
+            self.set_last_modified(last_modified)
+        if input_datasets is not None:
+            self.set_input_datasets(input_datasets)
+
+    @classmethod
+    def _new_from_graph(cls, urn: Urn, current_aspects: models.AspectBag) -> Self:
+        assert isinstance(urn, ChartUrn)
+        entity = cls(
+            platform=urn.dashboard_tool,
+            name=urn.chart_id,
+        )
+        return entity._init_from_graph(current_aspects)
+
+    @property
+    def urn(self) -> ChartUrn:
+        return self._urn  # type: ignore
+
+    def _ensure_chart_props(self) -> models.ChartInfoClass:
+        """Ensure chart properties exist, using a safer approach."""
```

Comment:
in other methods like this, we use `self._setdefault_aspect`
didn't work that here?